### PR TITLE
Start new http session for downloads without auto decompress

### LINKED ISSUE #159: Encoded content is decoded resulting in wrong size/checksum
If I understood it right, flatpak expects raw http data for `extra-data`, i.e. if it has `Content-Encoding`, it's not taken into account and size and checkum apply to undecoded content. Probably this behavior depends on libcurl/libsoup?
Our http client automatically decodes content if it detects `Content-Encoding`, so we calculate size/checksum for decoded content, which is wrong for flatpak.
On Flathub I've found only one app with `gzip`'ed upstream URL for extra-data - [GravitDesigner](https://github.com/flathub/io.designer.GravitDesigner).

### PR DESCRIPTION
**Pull request by gasinvein**
Fixes #159.

### CODE REVIEW

File: tests/test_checker.py [hunk 1]
```diff
@@ -437,11 +439,19 @@ async def test_check(self):
             relative_redirect.new_version.url,
             "https://httpbingo.org/base64/MzAtNTAgZmVyYWwgaG9ncyEK",
         )
-        self.assertEqual(
+        # XXX: Gzipped response will have size/checksum different from plain; match any
```
Comment (wjt) [inline tests/test_checker.py]: Isn't the intention of the patch to make these deterministically have the checksum of the compressed content? In which case, why does the test allow both?

Comment (gasinvein) [reply tests/test_checker.py]: We can't tell httpbingo to encode (or not) the content. Currently httbingo feeds us gzipped response, but I won't be surprised if it suddenly starts to feed plain, thus here check for both.

Comment (wjt) [reply tests/test_checker.py]: Can we put this in the comment?

Comment (gasinvein) [reply tests/test_checker.py]: Sure, done.

Comment (gasinvein) [reply tests/test_checker.py]: If you know where we can get a response with well-known body that is deterministically gzipped, we can add it to tests.

Comment (wjt) [reply tests/test_checker.py]: Probably the sustainable fix here is to put an HTTP server into the test suite and serve data locally. e.g. https://pytest-httpserver.readthedocs.io/en/latest/

**Review by wjt — APPROVED** [review]

**Pull request merged**